Mathlib/RingTheory/Valuation/Basic.lean

```diff
@@ -349,6 +349,28 @@ theorem one_lt_val_iff (v : Valuation K Γ₀) {x : K} (h : x ≠ 0) : 1 < v x 
   simpa using (inv_lt_inv₀ (v.ne_zero_iff.2 h) one_ne_zero).symm
 #align valuation.one_lt_val_iff Valuation.one_lt_val_iff
 
+theorem one_le_val_iff (v : Valuation K Γ₀) {x : K} (h : x ≠ 0) : 1 ≤ v x ↔ v x⁻¹ ≤ 1 := by
+  convert (one_lt_val_iff v (inv_ne_zero h)).symm.not <;>
+  push_neg <;> simp only [inv_inv]
+
+theorem val_lt_one_iff (v : Valuation K Γ₀) {x : K} (h : x ≠ 0) : v x < 1 ↔ 1 < v x⁻¹ := by
+  convert (one_lt_val_iff v (inv_ne_zero h)).symm
+  simp only [inv_inv]
+
+theorem val_le_one_iff (v : Valuation K Γ₀) {x : K} (h : x ≠ 0) : v x ≤ 1 ↔ 1 ≤ v x⁻¹ := by
+  convert (one_le_val_iff v (inv_ne_zero h)).symm
+  simp only [inv_inv]
+
+theorem val_eq_one_iff (v : Valuation K Γ₀) {x : K} (h : x ≠ 0) : v x = 1 ↔ v x⁻¹ = 1 := by
+  simpa only [le_antisymm_iff, And.comm] using and_congr (one_le_val_iff v h) (val_le_one_iff v h)
+
+-- This theorem is not stated in the strongest form but is very useful.
```

Comment:
And what would be its strongest form? Why not stating it in that generality?

Reply:
The strongest form is  `v x ≤ 1 ∨ v x⁻¹ < 1`. But based on my experience, many proofs just need you to discuss whether `x` or `x⁻¹` is an integer and the proof goes exactly the same afterwards (thus saves lots of lines if we use this form `v x ≤ 1 ∨ v x⁻¹ ≤ 1`). `v x ≤ 1 ∨ v x⁻¹ < 1` will make the `x⁻¹` case be treated differently with a small `le_of_lt`.

Reply:
Perhaps add both lemmas, then, with the same name (and one with a `'` )